refinery/file_store/models.py

```diff
@@ -399,6 +390,19 @@ def get_file_object(file_name):
         return None
 
 
+@receiver(post_save, sender=FileStoreItem)
+def _set_filetype(sender, instance, created, **kwargs):
+    """Set file type using file extension if no file type provided"""
+    if created and not instance.filetype:
+        try:
+            extension = instance.get_file_extension()
+        except RuntimeError as exc:
+            logger.warn("Could not assign type to file '%s': %s",
+                        instance, exc)
+        else:
+            instance.filetype = extension.filetype
```

Comment:
Wouldn't this instance need to `save()` again after updating this field? Or rather `update()` as to not trigger the signal again.

Maybe this is a better fit for the `pre_save` signal?